cpp/include/cudf/copying.hpp

```diff
@@ -80,6 +80,70 @@ std::unique_ptr<column> allocate_like(column_view input, size_type size,
  */
 std::unique_ptr<table> empty_like(table_view input_table);
 
+/**
+ * @brief Copies a range of elements in-place from one column to another.
+ *
+ * Overwrites the range of elements in @p output indicated by the indices
+ * [@p out_begin, @p out_ned) with the elements from @p input indicated by the
+ * indices [@p in_begin, @p in_begin + N) (where N =
+ * (@p out_end - @p out_begin)). Use the out-of-place copy function returning
+ * std::unique_ptr<column> for uses cases requiring memory reallocation.
+ *
+ * If @p input and @p output refer to the same elements and the ranges overlap,
+ * the behavior is undefined.
+ *
+ * @throws `cudf::logic_error` if memory reallocation is required (e.g. for
+ * variable width types).
+ * @throws `cudf::logic_error` for invalid range (if @p out_begin < 0,
+ * @p out_begin > @p out_end, @p out_begin >= @p output.size(),
+ * @p out_end > @p output.size(), @p in_begin < 0, in_begin >= @p input.size(),
+ * or @p in_begin + @p out_end - @p out_begin > @p input.size()).
+ * @throws `cudf::logic_error` if @p output and @p input have different types.
+ * @throws `cudf::logic_error` if @p input has null values and @p output is not
+ * nullable.
+ *
+ * @param output The preallocated column to copy into
+ * @param input The column to copy from
+ * @param out_begin The starting index of the output range (inclusive)
+ * @param out_end The index of the last element in the output range (exclusive)
+ * @param in_begin The starting index of the input range (inclusive)
+ * @return void
+ */
+void copy_range(mutable_column_view& output, column_view const& input,
```

Comment:
Question:  Should this take a cudaStream_t?  While we are not externally exposing different streams, this is a utility function that could be called from something that does know what stream it wants to execute on.

Reply:
AFAIK, to pass the stream argument, you need to call the detail space function (not the public API).

They are defined in

cpp/include/cudf/detail/copy_range.cuh